adal/src/main/java/com/microsoft/aad/adal/AcquireTokenSilentHandler.java

```diff
@@ -269,7 +288,7 @@ private AuthenticationResult acquireTokenWithCachedItem(final TokenCacheItem cac
             throws AuthenticationException {
         final AuthenticationResult result = acquireTokenWithRefreshToken(cachedItem.getRefreshToken());
         
-        if (result != null) {
+        if (result != null && !result.isExtendedLifeTimeToken()) {
```

Comment:
Let's say in the scenario where retry fails for the server_not_responding. In acquireTokenWithRefreshToken(), we already get the stale AT back. So here result is not null and because it is a stale AT, we do not want to update the cache. This is the reason why I did the check here. 

Thanks.